### PR TITLE
Elizabeth solution

### PR DESCRIPTION
**Pull request by elizabethvenner**

### CODE REVIEW

File: js/notes_for_mary.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+I also think I could have benefited from judicious use of switch statements for some of the functions.
+
+Other than this, I have not set the MAXQUALITY for backstage pass and think might need to do a few more tests to clear up edge cases. Think could also have created a few more constants.
+
+Tests could also be neater.
+
+I do have a couple of questions/things I would like you to look at:
+
+1) I know it's quite popular to refactor e.g. line 30 (and in fact most of my conditional statements) into a function of their own which would then be called inside the if condition, but when there are so many, I wonder if this would make it less readable as having to look further down the page to see what the function does (though appropriate naming would certainly help here)
```
Comment (maryrosecook) [inline js/notes_for_mary.md:11]: In the vast majority of cases, a clearly named function is better than inlined code.  The name communicates what the code does more clearly and more tersely.

function

File: js/notes_for_mary.md [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+2) Aged Brie - I didn't update the sellIn, mainly because it seemed irrelevant as it doesn't affect the quality. Also, logically, it seems that in this case there would be no sellIn, as there is not a date by which it would have to be sold?
```
Comment (maryrosecook) [inline js/notes_for_mary.md:13]: That seems very reasonable.

File: js/notes_for_mary.md [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+3) Is my updateQuality method ind of like what you meant when you were mentioning how both Felix's and Stuart's code could be refactored to avoid a massive switch statement/lots of ifs? Though I do still have the problem where I need to add to both special_items as well as a function. Perhaps the function could have been defined in the special_items object? I did look at doing this but that seemed to create other problems.
```
Comment (maryrosecook) [inline js/notes_for_mary.md:15]: The updateQuality is better abstracted than the code that Felix and Stuart showed.  The selection of the right updateQuality function is mechanical, so it's easier to add new cases, clearer, and harder to mess up than using an if statement.  You could take it a step further by changing the regular update item to be just another case in the updateQuality lookup object you created.

The inheritance approach is another way of achieving the same thing.  It's a little cleaner because you can put each special updateQuality code in its own file.

**Review by maryrosecook — COMMENTED** [review]
Here you go!
